eng/native/build-commons.sh

```diff
@@ -66,8 +46,16 @@ check_prereqs()
     fi
 }
 
+copy_version_files()
+{
+    # Copy dummy version files if version files haven't already been generated
+    cp -r -n "$__RepoRootDir/eng/native/version/"* "$__RepoRootDir/artifacts/obj/"
+}
+
 build_native()
 {
+    eval "$__RepoRootDir/eng/native/version/copy_version_files.sh"
```

Comment:
Should we just inline the `cp` line (as done on line 52) rather than indirection and eval? Note that we don't always have 1:1 correspondence between .sh and .cmd, when it comes to internal scripts.

Reply:
I prefer to move towards a 1:1 correspondence to simplify maintenance costs.

Reply:
Separate file is alright, but concern was regarding usage of `eval`: in other places like this file's (`build-commons.sh`) own usage, we have managed to avoid eval by using sourcing. eval is generally considered a last resort in scripting languages. We can use `source copy_version_files.sh` (or `. copy_version_files.sh`) syntax here as well.

This is my ideal version version of `copy_version_files.sh`, consistent with most of the existing code:

```sh
#!/usr/bin/env bash

# `source` this file, as we do for `build-common.sh`'s call-sites.
# i.e. __RepoRootDir etc. are already defined in caller scripts.

__VersionFolder="$__RepoRootDir/eng/native/version"

for path in "${__VersionFolder}/"*{.h,.c}; do
    if [ ! -s "$__RepoRootDir/artifacts/obj/$(basename "$path")" ]; then
        cp "$path" "$__RepoRootDir/artifacts/obj/"

        if [[ "$path" == *_version.c ]]; then
            # update commit
            commit="$(git rev-parse HEAD 2>/dev/null)"
            commit="${commit:-N/A}"
            substitute="$(printf 'static char sccsid[] __attribute__((used)) = "@(#)Version N/A @Commit: %s";\n' "$commit")"
            sed -i '' "s|^static.*|$substitute|" "$__RepoRootDir/artifacts/obj/_version.c"
        fi
    fi
done
```

Reply:
For testing purposes, it's very nice to be able to manually call this script without having to set up any environment variables beforehand. I'd like to keep that experience if possible. If there's a way to call into another script using the current scripting engine when the path is constructed from vars in bash, I'd happily switch to that from `eval`.